Recent libgit versions have added struct members to git_fetch_options
and git_remote_callbacks which change the in-memory layout of these
structs (e.g. the ABI).

This patch adds these members to the struct layouts specified in
guile-git. Thereby fixing compatiblity with these recent libgit
versions.

diff -upr guile-git-v0.5.2.orig/git/structs.scm guile-git-v0.5.2/git/structs.scm
--- guile-git-v0.5.2.orig/git/structs.scm	2023-08-03 22:59:18.830378658 +0200
+++ guile-git-v0.5.2/git/structs.scm	2023-08-03 22:59:58.957149650 +0200
@@ -637,6 +637,7 @@ type to 'specified for this to take effe
                (push-update-reference ,(bs:pointer uint8))
                (push-negotiation ,(bs:pointer uint8))
                (transport ,(bs:pointer uint8))
+               (remote-ready ,(bs:pointer uint8))
                (payload ,(bs:pointer uint8))
 
                ;; libgit2 1.0 added this field, which is missing from 0.28.5,
@@ -670,6 +671,8 @@ type to 'specified for this to take effe
                (update-fetchhead ,int)
                (download-tags ,int)
                (proxy-opts ,%proxy-options)
+               (depth ,int)
+               (follow-redir ,int)
                (custom-headers ,%strarray))))
 
 (define-record-type <fetch-options>
